documentation/experimental_features.md

```diff
@@ -8,6 +8,99 @@ API may change, requiring the user to update their code or configuration. The
 Bolt team attempts to make these changes painless by providing useful warnings
 around breaking behavior where possible. 
 
+## `Parallelize()` function
+
+This feature was introduced in [Bolt
+2.35.0](https://github.com/puppetlabs/bolt/tree/main/CHANGELOG.md#bolt-2350-2020-11-16).
+
+Bolt plan functions have always run concurrently across targets - that is, if a function takes a
+list of targets and operates on them it runs that step on each target in parallel. For example, the
+following plan will first run `hostname` on all targets at the same time, wait for all targets to
+finish, then will run `whoami` on all targets at the same time.
+
+```
+# $targets = target1,target2,target3
+plan myplan(TargetSpec $targets) {
+  run_command('hostname', $targets)
+  run_command('whoami', $targets)
+}
+```
+
+The experimental `parallelize()` function accepts an array and a block, and runs the entire block on
+each array element in parallel. In the example above, target3 will have to wait for `hostname` to
+finish on target1 and target2 before it can run `whoami`. Inside a parallelize block targets will be
+able to run subsequent plan functions before all targets have finished each step. Let's see the same
+plan with a parallelize block:
```

Comment:
```suggestion
In the example above, `target3` has to wait for `hostname` to finish on `target1` and `target2` 
before it can run `whoami`. The experimental `parallelize()` function accepts an array and a block, 
and runs the entire block on each array element in parallel. Inside a parallelize block, targets can
run subsequent plan functions before all targets have finished each step. For example, here is the 
same plan with a parallelize block:
```